src/index.ts

```diff
@@ -388,6 +388,10 @@ async function getClassifiedViaPostMessage(): Promise<string> {
       window.addEventListener("message", listen)
 
       function listen(event: MessageEvent<string>) {
+        if (new URL(event.origin).host !== new URL(setup.endpoints.lobby).host) {
+          console.log(`Got a message from ${event.origin} while waiting for login credentials. Ignoring.`)
+          return
+        }
         window.removeEventListener("message", listen)
         if (event.data) {
           resolve(event.data)
```

Comment:
In addition to this, we could also easily check if this is the correct data?
Check if it's JSON and if the object has a `sessionKey` property.

Reply:
Yeah. I like this. I've added a commit for that :+1: